sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2001,6 +2001,7 @@ class Dataset[T] private[sql](
   * @group typedrel
   * @since 3.0.0
   */
+  @varargs
```

Comment:
LGTM. Just want to know which is a new specification add this annotation?

Reply:
We should add this whenever it's supposed to be Java API if it uses `*` args (https://github.com/databricks/scala-style-guide#java-varargs)

Reply:
OK. I got it.